ESP8266Interface.cpp

```diff
@@ -507,6 +507,23 @@ nsapi_error_t ESP8266Interface::setsockopt(nsapi_socket_t handle, int level,
     return NSAPI_ERROR_UNSUPPORTED;
 }
 
+nsapi_error_t ESP8266Interface::getsockopt(nsapi_socket_t handle, int level, int optname, void *optval, unsigned *optlen)
+{
+    struct esp8266_socket *socket = (struct esp8266_socket *)handle;
+
+    if (level == NSAPI_SOCKET) {
+        switch (optname) {
+            case NSAPI_KEEPALIVE: {
+                *optlen = (unsigned)sizeof(int);
```

Comment:
Technically `getsockopt` is supposed to truncate its output into the supplied buffer length.

That would be a final "copy out" memcpy stage applying to all options, as seen in Nanostack's `socket_getsockopt`. Not terribly relevant for this single-integer option, but maybe should be put in as a "correct practice" framework? I note the NSAPI header doesn't specify that POSIX behaviour though.

I also note lwIP's glue doesn't implement `getsockopt` at all either, so clearly no-one cares about this much.

Reply:
ok, good to know.